crc/api/workflow.py

```diff
@@ -161,17 +112,20 @@ def set_current_task(workflow_id, task_id):
     user_uid = __get_user_uid(workflow_model.study.user_uid)
     processor = WorkflowProcessor(workflow_model)
     task_id = uuid.UUID(task_id)
-    task = processor.bpmn_workflow.get_task(task_id)
-    if task.state != task.COMPLETED and task.state != task.READY:
+    spiff_task = processor.bpmn_workflow.get_task(task_id)
+    if spiff_task.state != spiff_task.COMPLETED and spiff_task.state != spiff_task.READY:
         raise ApiError("invalid_state", "You may not move the token to a task who's state is not "
                                         "currently set to COMPLETE or READY.")
 
     # Only reset the token if the task doesn't already have it.
-    if task.state == task.COMPLETED:
-        task.reset_token(reset_data=False)  # we could optionally clear the previous data.
+    if spiff_task.state == spiff_task.COMPLETED:
+        spiff_task.reset_token(reset_data=True)  # Don't try to copy the existing data back into this task.
```

Comment:
We should always reset the data - clear it out so that the task data is in the state it was in when this task was first executed.